sdk/keyvault/azure-security-keyvault-secrets/test/ut/secret_client_test.cpp

```diff
@@ -52,3 +50,278 @@ TEST(SecretClient, GetUrl)
+    EXPECT_EQ(secret.Name, secretName);
+  }
+  {
+    auto secretResponse = client.SetSecret(secretName, "secretValue2");
+    CheckValidResponse(secretResponse);
+    auto secret = secretResponse.Value;
+    version2 = secret.Properties.Version;
+    EXPECT_EQ(secret.Name, secretName);
+  }
+  {
+    auto secretResponse = client.GetPropertiesOfSecretsVersions(secretName);
+    EXPECT_EQ(secretResponse.Items.size(), size_t(2));
+    EXPECT_EQ(secretResponse.Items[0].Version, version1);
+    EXPECT_EQ(secretResponse.Items[1].Version, version2);
+  }
+  {
+    auto operation = client.StartDeleteSecret(secretName);
+    operation.PollUntilDone(1s);
+    auto deletedSecretResponse = client.GetDeletedSecret(secretName);
+    CheckValidResponse(deletedSecretResponse);
+    auto secret = deletedSecretResponse.Value;
+    EXPECT_EQ(secret.Name, secretName);
+  }
+  {
+    auto purgedResponse = client.PurgeDeletedSecret(secretName);
+    CheckValidResponse(purgedResponse, Azure::Core::Http::HttpStatusCode::NoContent);
+  }
+}
+
+TEST_F(KeyVaultSecretClientTest, UpdateTest)
+{
+  auto secretName = "CreateSecretWithThisName";
+  SecretProperties properties;
+  auto const& client = GetClientForTest(
+      ::testing::UnitTest::GetInstance()->current_test_info()->name(),
+      Azure::Core::Test::TestMode::PLAYBACK);
+
+  {
+    auto secretResponse = client.SetSecret(secretName, "secretValue");
+    CheckValidResponse(secretResponse);
+    auto secret = secretResponse.Value;
+    EXPECT_EQ(secret.Name, secretName);
+  }
+  {
+    // Now get the key
+    auto secretResponse = client.GetSecret(secretName);
+    CheckValidResponse(secretResponse);
+    auto secret = secretResponse.Value;
+    properties = secret.Properties;
+    EXPECT_EQ(secret.Name, secretName);
+  }
+  {
+    properties.ContentType = "xyz";
+    UpdateSecretPropertiesOptions options;
+    auto secretResponse = client.UpdateSecretProperties(secretName, properties.Version, properties);
+    CheckValidResponse(secretResponse);
+    auto secret = secretResponse.Value;
+    EXPECT_EQ(secret.Name, secretName);
+    EXPECT_EQ(secret.Properties.ContentType.Value(), properties.ContentType.Value());
+  }
+  {
+    auto operation = client.StartDeleteSecret(secretName);
+    operation.PollUntilDone(1s);
+    auto deletedSecretResponse = client.GetDeletedSecret(secretName);
+    CheckValidResponse(deletedSecretResponse);
+    auto secret = deletedSecretResponse.Value;
+    EXPECT_EQ(secret.Name, secretName);
+  }
+  {
+    auto purgedResponse = client.PurgeDeletedSecret(secretName);
+    CheckValidResponse(purgedResponse, Azure::Core::Http::HttpStatusCode::NoContent);
+  }
+}
+
+TEST_F(KeyVaultSecretClientTest, BackupRestore)
+{
+  auto secretName = "CreateSecretWithThisName";
+  std::vector<uint8_t> backupData;
+  auto const& client = GetClientForTest(
+      ::testing::UnitTest::GetInstance()->current_test_info()->name(),
+      Azure::Core::Test::TestMode::PLAYBACK);
```

Comment:
The test should run without telling the `TestMode`.  It should only pass the test name and use whatever is in the env to run the test.

If you are doing this for CI, the strategy for that is a little different. We might want to update the CI config to set the env var to PLAYBACK for whatever CI pipeline we want.

The problem is that right now, using the current approach, we can't switch to RECORD or LIVE for these tests. They are updating the env to always run on playback, no matter what.

Reply:
this is more convenient to run the test , first you write the test and pass record, then pass playback

Reply:
The should not know about recording vs playback. 

Add the required env vars for CI here: https://github.com/Azure/azure-sdk-for-cpp/blob/main/eng/pipelines/templates/jobs/archetype-sdk-client.yml#L134

If the TEST_MODE env var is not set, the test is expected to run LIVE mode

Reply:
from the point of view of the test it should run the same irrespective of where the test is run., i should have predictable results irrespective of the environment i am running in. I cannot rely on CI config , which also changes over time , to ensure my tests run and give the expected results when test mode can shift at any time due to the interference of external entities.